drivers/pinmux/stm32/pinmux_stm32.c

```diff
@@ -116,6 +116,103 @@ static int stm32_pin_configure(int pin, int func, int altf)
 				    STM32_PIN(pin), func, altf);
 }
 
+/**
+ * @brief helper for converting dt stm32 pinctrl format to existing pin config
+ *        format
+ *
+ * @param *pinctrl pointer to soc_gpio_pinctrl list
+ * @param list_size list size
+ */
+
+void stm32_dt_pinctrl_configure(const struct soc_gpio_pinctrl *pinctrl,
+				size_t list_size)
+{
+	struct device *clk;
+	uint32_t pin, mux;
+	uint32_t func = 0;
+
+	/* make sure to enable port clock first */
+	clk = device_get_binding(STM32_CLOCK_CONTROL_NAME);
+
+	for (int i = 0; i < list_size; i++) {
+		mux = pinctrl[i].pinmux;
+
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32f1_pinctrl)
```

Comment:
Would it be possible to write this like this
```
if (DT_HAS_COMPAT_STATUS_OKAY(st_stm32f1_pinctrl)) {
...
} else {

}
```
which would let the compiler to verify both branches?

Reply:
Actually we don't use the same defines in both branches and compilation would fail due to missing inclusions